homeassistant/components/nightscout/sensor.py

```diff
@@ -120,3 +162,61 @@ def _parse_icon(self) -> str:
     def extra_state_attributes(self):
         """Return the state attributes."""
         return self._attributes
+
+
+class Battery(CoordinatorEntity, SensorEntity):
+    """Battery sensor of Nightscout device."""
+
+    # SENSOR_ATTR = "battery_percentage_remaining"
+    _attr_device_class = DEVICE_CLASS_BATTERY
+    _attr_state_class = STATE_CLASS_MEASUREMENT
+    _attr_native_unit_of_measurement = PERCENTAGE
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
+
+    def __init__(self, coordinator, api: NightscoutAPI, name, unique_id):
+        """Initialize the Nightscout sensor."""
+        super().__init__(coordinator)
+        self.api = api
+        self._unique_id = unique_id
+        self._name = name
+        self._device_name = name
+        self._icon = "mdi:battery-unknown"
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the sensor."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def available(self):
+        """Return if the sensor data are available."""
+        return hasattr(self.coordinator.data[self._name], "uploader")
+
+    @property
+    def native_value(self):
+        """Return the state of the device."""
+        return self.coordinator.data[self._name].uploader.battery
+
+    @property
+    def icon(self) -> str:
+        """Battery state icon handling."""
+        return icon_for_battery_level(
+            battery_level=self.coordinator.data[self._name].uploader.battery,
+            charging=False,
+        )
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        uploader = self.coordinator.data[self._name].uploader
+        attr = {}
+        if hasattr(uploader, "type"):
+            attr[ATTR_TYPE] = uploader.type
```

Comment:
What kind of value is this? Please give an example.

Reply:
Nightscout's documentation lacks the possible values... but in my case, my MiaoMiao collector is a `BRIDGE` and my phone running xDrip+ is a `PHONE`:
```json
[
  {
    "_id": "61ce18f4972af90182e5f5d4",
    "device": "Tomato",
    "uploader": {
      "battery": 80,
      "type": "BRIDGE"
    },
    "created_at": "2021-12-30T20:39:16.035Z",
    "utcOffset": -180,
    "mills": 1640896756035
  },
  {
    "_id": "61ce18f3972af90182e5f5d3",
    "device": "samsung SM-N986B",
    "uploader": {
      "battery": 72,
      "type": "PHONE"
    },
    "created_at": "2021-12-30T20:39:15.988Z",
    "utcOffset": -180,
    "mills": 1640896755988
  }
]
```

Reply:
Do we need to add this info to state attributes? It's not dynamic, right? State attributes is for dynamic info related to the state of the entity.

We should probably set the `device_info` property on the battery entities since they represent measurements from different devices.

Reply:
that makes sense! I'll update the PR